### PR TITLE
Integration with wordpress API

### PR DESCRIPTION
**Pull request by Peter-holyoke**
## Related Issue

https://github.com/ProvidenceGeeks/website-lambda-wordpress/issues/5

## Summary of Changes

Now pulls blog posts from the Wordpress API at http://blog.pvdgeeks.org/wp-json/wp/v2/posts

Also added release script from website-lambda-meetup repo and modified to fit this repo.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,6 +11,11 @@
   "author": "pholyoke@kenzan.com",
   "license": "MIT",
   "devDependencies": {
-    "eslint": "^4.8.0"
+    "archiver": "^2.1.0",
+    "eslint": "^4.8.0",
+    "rimraf": "^2.6.2"
+  },
+  "dependencies": {
+    "aws-sdk": "^2.154.0"
```
Comment (thescientist13) [inline package.json]: not a big deal, but I like to order `dependencies` before `devDependencies`

File: src/index.js [hunk 2]
```diff
@@ -1,2 +1,106 @@
...
\ No newline at end of file
+// expose handler for Lambda
+exports.run = run;
+
+if (!isProduction) {
+  run();
+}
+
+function run() {
+  const resolveAllPromises = isProduction ? resolveBlogPostsDataS3 : resolveBlogPostsDataLocal;
```
Comment (thescientist13) [inline src/index.js]: if we're only make one request, then the name should probably change away from a plural, right?  So maybe something like `promiseResolver`

File: src/index.js [hunk 3]
```diff
@@ -1,2 +1,106 @@
...
+  const blogPosts = getBlogPostsData();
+
+  blogPosts
```
Comment (thescientist13) [inline src/index.js]: we should just be able to pass a function reference right to the `then` function, like we do with `catch`, e.g.
```
getBlogPostsData()
  .then(resolveAllPromises)
  .catch(handleError);
```

**Review by thescientist13 — COMMENTED** [review]
good job @Peter-holyoke !

Left you some feedback, let me know your thoughts.

File: package.json (hunk 1 above)
Comment (Peter-holyoke) [reply package.json]: Not sure why this happened. I never manually edited it to be this way, I'll fix it though.

File: src/index.js [hunk 4]
```diff
@@ -1,2 +1,108 @@
+/*
+ * This script assumes AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are exported as environment variables
+ */
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const http = require('http');
+const isProduction = process.env.NODE_ENV === 'production';
+const outputFile = 'blog-posts-data.json';
```
Comment (thescientist13) [inline src/index.js]: let's call it _wordpress-data.json_ for now, so it's easy to see in S3 where the source of the data came from specifically.

File: src/index.js [hunk 5]
```diff
@@ -1,2 +1,108 @@
...
+exports.run = run;
+
+if (!isProduction) {
+  run();
+}
+
+function run() {
+  const promiseResolver = isProduction ? resolveBlogPostsDataS3 : resolveBlogPostsDataLocal;
+
+  getBlogPostsData()
```
Comment (thescientist13) [inline src/index.js:24]: I'm pretty sure we can just pass our promise right to `then`, e.g.
```
.then(promiseResolver)
```

File: src/index.js [hunk 6]
```diff
@@ -1,2 +1,108 @@
...
+ * This script assumes AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are exported as environment variables
+ */
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const http = require('http');
+const isProduction = process.env.NODE_ENV === 'production';
+const outputFile = 'blog-posts-data.json';
+const s3Config = {
+  bucket: 'data.pvdgeeks.org',
+  key: 'blogpost'
```
Comment (thescientist13) [inline src/index.js]: change the `key` to _wordpress_

File: release.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
+/*
+ * This script assumes AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY are exported as environment variables
+ */
+const archiver = require('archiver');
+const AWS = require('aws-sdk');
+const fs = require('fs');
+const zipOutput = './archive/blog-posts.zip';
```
Comment (thescientist13) [inline release.js]: I know these changes came a bit after this, but's lets match the implementation here, and change the name to _wordpress_
https://github.com/ProvidenceGeeks/website-lambda-meetup/blob/master/release.js#L9
```
const archiveOutputDirectory = './archive';
const s3Bucket = 'lambda.pvdgeeks.org';
const s3Key = `wordpress-${packageJson.version}.zip`;
const zipOutput = `./archive/${s3Key}`;
```

Also note the change to the `release` function
https://github.com/ProvidenceGeeks/website-lambda-meetup/blob/master/release.js#L40

**Review by thescientist13 — COMMENTED** [review]
Just some minor follow up comments @Peter-holyoke and apologies on the late decision to rename it to _wordpress_ instead of _blog-posts_

**Review by thescientist13 — APPROVED** [review]

**Pull request merged**